packages/block-library/src/rss/index.php

```diff
@@ -5,6 +5,46 @@
  * @package WordPress
  */
 
+/**
+ * Helper function to trim text to a certain number of characters while preserving whole words.
+ *
+ * @param string $to_trim The text to trim.
+ * @param int    $character_count The maximum number of characters to return.
+ * @param string $more The string to append to the trimmed text.
+ */
+function trim_to_word_boundary( $to_trim, $character_count, $more = ' [&hellip;]' ) {
+	// Strip tags to mimic behavior of `wp_trim_words`.
+	$to_trim = wp_strip_all_tags( $to_trim );
+
+	if ( strlen( $to_trim ) <= $character_count ) {
+		return $to_trim;
+	}
+
+	preg_match( '/^.{0,' . $character_count . '}\s/', $to_trim, $parts );
```

Comment:
This function fails if a pass a string without spaces:

```php
$string = 'loremipsumloremipsumloremipsumloremipsumloremipsumloremipsumloremipsum';
echo trim_to_word_boundary($string, 20);
```

Output:
```
PHP Fatal error:  Uncaught ErrorException: Undefined array key 0 in /path_to_wordpress/test.php:25
Stack trace:
#0 /path_to_wordpress/test.php(25): {closure}(2, 'Undefined array...', '/Users/anton/ht...', 25)
#1 /path_to_wordpress/test.php(30): trim_to_word_boundary('loremipsumlorem...', 20)
#2 {main}
  thrown in /path_to_wordpress/test.php on line 25
```

Reply:
good catch!